frontend/ui-picasso/app/components/Molecules/AllBondsTable.tsx

```diff
@@ -14,17 +15,46 @@ import { TokenAsset, TokenPairAsset } from "@/components";
 import { BondOffer } from "@/stores/defi/polkadot/bonds/types";
 import { getROI } from "@/defi/polkadot/pallets/BondedFinance";
 import { humanBalance } from "@/utils/formatters";
+import { useQuery } from "@apollo/client";
+import { GET_BONDED_FINANCE } from "@/apollo/queries";
+import BigNumber from "bignumber.js";
 
 export type AllBondsTableProps = TableContainerProps & {
   bonds?: BondOffer[];
   onRowClick: (offerId: string) => void;
 };
 
+function getTotalPurchasedInFormat(
+  currentBond: {
+    totalPurchased: string;
+  },
+  bondPrice: BigNumber,
+  price: BigNumber
+) {
+  let totalPurchased: number | string = 0;
+  if (currentBond) {
+    totalPurchased = currentBond.totalPurchased;
+  }
+  return humanBalance(
+    new BigNumber(totalPurchased)
+      .multipliedBy(bondPrice)
+      .multipliedBy(price)
+      .toString()
+  );
+}
+
 export const AllBondsTable: React.FC<AllBondsTableProps> = ({
   bonds,
   onRowClick = () => {},
   ...rest
 }) => {
+  const { loading, data, error } = useQuery(GET_BONDED_FINANCE);
+
+  if (loading || error) {
+    return <Skeleton width={200} height={50} />;
+  }
```

Comment:
Maybe we should handle the error separately, as otherwise an error will look like if it is still loading